src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorPreCompileModule.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Versioning;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.Dnx.Compilation.CSharp;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.DependencyInjection;
+
+namespace Microsoft.AspNet.Mvc.Razor.Precompilation
+{
+    /// <summary>
+    /// An <see cref="ICompileModule"/> implementation that pre-compiles Razor views in the application.
+    /// </summary>
+    public abstract class RazorPreCompileModule : ICompileModule
+    {
+        private readonly IAssemblyLoadContext _loadContext;
+        private readonly object _memoryCacheLookupLock = new object();
+        private readonly Dictionary<PrecompilationCacheKey, MemoryCache> _memoryCacheLookup =
+            new Dictionary<PrecompilationCacheKey, MemoryCache>();
+
+        /// <summary>
+        /// Instantiates a new <see cref="RazorPreCompileModule"/> instance.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceProvider"/> for the application.</param>
+        public RazorPreCompileModule(IServiceProvider services)
+        {
+            _loadContext = services.GetRequiredService<IAssemblyLoadContext>();
```

Comment:
That should go away once this PR goes in - https://github.com/aspnet/Mvc/pull/2961